preprocessing.py

```diff
@@ -33,9 +34,21 @@ def one_doc_preprocess(ar, list_of_arguments, dictionary_of_all_terms):
                 if word not in stop_words:
                     print(ps.stem(word), file=output_file)
                     if word not in dictionary_of_all_terms.keys():
-                        dictionary_of_all_terms[num_of_keys] = word
+                        dictionary_of_all_terms[ps.stem(word)] = num_of_keys
                         num_of_keys += 1
 
+def sparse_basis_vector_creation(i, n):
```

Comment:
I think that this method  and "creation_sparse_index" is excess 
see comments below